### PR TITLE
add onprem consumer

### PR DESCRIPTION
**Pull request by rccrdpccl**
This change adds a new type of consumer: onprem event parser. Its aim is to read events from the ingress' announce topic, download the tarball when we found relevant messages, parse it and hydrate with new events the same pipeline as it's currently used by SaaS

### CODE REVIEW

**Comment by gamli75** [comment]
@rccrdpccl I saw only one line with logger.Info :) - can you add some more logs to this code? which cluster id? how many events? maybe it will be helpful to find issues in the service/ process

File: internal/onprem/hydrate.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+	"github.com/openshift-assisted/assisted-events-streams/internal/types"
+	"github.com/openshift-assisted/assisted-events-streams/pkg/stream"
+	kafka "github.com/segmentio/kafka-go"
+	"github.com/sirupsen/logrus"
+)
+
+const internalDownloadBuffer int = 1000
+const internalUntarBuffer int = 1000
+const internalEventBuffer int = 1000
```
Comment (CrystalChun) [inline internal/onprem/hydrate.go:16]: Small: maybe it'd be nice to have a comment about how these numbers were determined?

File: pkg/stream/reader.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ type KafkaConfig struct {
 	SaslMechanism   string `envconfig:"KAFKA_SASL_MECHANISM" default:"PLAIN"`
 	Topic           string `envconfig:"KAFKA_EVENT_STREAM_TOPIC" required:"true"`
 	GroupID         string `envconfig:"KAFKA_GROUP_ID" required:"true"`
+	DstTopic        string `envconfig:"KAFKA_EVENT_STREAM_TOPIC_DST" required:"true"`
```
Comment (CrystalChun) [inline pkg/stream/reader.go:42]: Does `Dst` stand for "Destination"?

File: internal/types/event.go [hunk 3]
```diff
@@ -1,5 +1,10 @@
 package types
 
+type EventEnvelope struct {
+	Event Event
+	Key   []byte
```
Comment (CrystalChun) [inline internal/types/event.go:5]: Will the `Key` always be the cluster id?

Comment (rccrdpccl) [reply internal/types/event.go:5]: The message Key will *always* be cluster id. The reason is that the topic we are sending to is split into N partitions (6), and the messages get sharded in function of the Key: a message with a given key will always end up in the same partition. Partitioning allow us to leverage consumer groups for faster processing, using cluster ID as key ensure we maintaing order among messages of the same cluster.

File: internal/onprem/hydrate.go (hunk 1 above)
Comment (rccrdpccl) [reply internal/onprem/hydrate.go:16]: Those were eyeballed but I can set them as env var.

File: pkg/stream/reader.go (hunk 2 above)
Comment (rccrdpccl) [reply pkg/stream/reader.go:42]: Yes it does, I will change to the full name

File: internal/types/event.go (hunk 3 above)
Comment (CrystalChun) [reply internal/types/event.go:5]: Nice, thank you for the explanation!

**Review by CrystalChun — APPROVED** [review]
/lgtm Thanks Riccardo!

**Review by adriengentil — APPROVED** [review]

**Pull request merged**